lighter/system.py

```diff
@@ -139,16 +139,30 @@ def forward(self, input: Union[torch.Tensor, List, Tuple]) -> Union[torch.Tensor
         Returns:
             Union[torch.Tensor, List, Tuple]: output of the model.
         """
+        # Keyword arguments to pass to the forward method
         kwargs = {}
         if hasarg(self.model.forward, "epoch"):
+            # Add `epoch` argument if forward accepts it
             kwargs["epoch"] = self.current_epoch
         if hasarg(self.model.forward, "step"):
+            # Add `step` argument if forward accepts it
             kwargs["step"] = self.global_step
 
-        if isinstance(input, list):
+        # Type not supported.
+        if not isinstance(input, (torch.Tensor, tuple, list, dict)):
+            logger.error(f"Input type '{type(input)}' not supported.")
+            sys.exit()
+
+        # Unpack Tuple or List. Only if num of args passed is less than or equal to num of args accepted.
+        if isinstance(input, (tuple, list)) and (len(input) + len(kwargs)) <= countargs(self.model):
             return self.model(*input, **kwargs)
+        # Unpack Dict. Only if dict's keys match criterion's keyword arguments.
+        elif isinstance(input, dict) and all([hasarg(self.model, name) for name in input]):
+            return self.model(**input, **kwargs)
+        # Tensor, List, or Dict, as-is, not unpacked.
         return self.model(input, **kwargs)
```

Comment:
Can we add this to `else:` ?

Makes it more readable IMO since we have two a lot of conditionals here.